centreon/www/front_src/src/Resources/Actions/Resource/Check/ListCheckOptions.tsx

```diff
@@ -0,0 +1,169 @@
+import { makeStyles } from 'tss-react/mui';
+
+import IconCheck from '@mui/icons-material/CheckOutlined';
+import List from '@mui/material/List';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import Popover from '@mui/material/Popover';
+import Divider from '@mui/material/Divider';
+
+import {
+  labelCheck,
+  labelCheckDescription,
+  labelForcedCheck,
+  labelForcedCheckDescription
+} from '../../../translatedLabels';
+
+import Text from './Text';
+
+const useStyles = makeStyles()((theme) => ({
+  button: {
+    alignItems: 'flex-start'
+  },
+  container: {
+    backgroundColor: theme.palette.background.default,
+    display: 'flex',
+    flexDirection: 'column',
+    minWidth: theme.spacing(13.5),
+    padding: theme.spacing(0, 1)
+  },
+  icon: {
+    minWidth: theme.spacing(3)
+  },
+  itemText: {
+    '& .MuiTypography-root': {
+      width: '100%'
+    },
+    display: 'flex',
+    flexDirection: 'row-reverse',
+    margin: 0,
+    paddingRight: theme.spacing(0.5)
+  }
+}));
+
+interface Disabled {
+  disableCheck: boolean;
+  disableForcedCheck: boolean;
+}
+
+enum VerticalEnum {
+  bottom = 'bottom',
+  center = 'center',
+  top = 'top'
+}
+
+enum HorizontalEnum {
+  center = 'center',
+  left = 'left',
+  right = 'right'
+}
+
+interface AnchorOrigin {
+  horizontal: HorizontalEnum;
+  vertical: VerticalEnum;
+}
+
+interface TransformOrigin {
+  horizontal: HorizontalEnum;
+  vertical: VerticalEnum;
+}
+
+interface Props {
+  anchorEl?: HTMLElement | null;
+  anchorOrigin?: AnchorOrigin;
+  disabled: Disabled;
+  isDefaultChecked: boolean;
+  onClickCheck: () => void;
+  onClickForcedCheck: () => void;
+  onClose: () => void;
+  open: boolean;
+  transformOrigin?: TransformOrigin;
```

Comment:
- I personally would not use enums to describe those positions values, I don't really see any added value. 
- AnchorOrigin & TransformOrigin are the same interface

so I would probably simplify everything by doing something like : 

```ts
interface PositionOrigin {
  horizontal: 'center' | 'left' | 'right';
  vertical: 'center' | 'bottom' | 'top';
}
```

```ts
const defaultAnchorOrigin = {
   horizontal: 'left',
   vertical: 'bottom'
 };

 const defaultTransformOrigin = {
   horizontal: 'left',
   vertical: 'top'
 };
```

```ts
  anchorOrigin?: PositionOrigin;
  disabled: Disabled;
  isDefaultChecked: boolean;
  onClickCheck: () => void;
  onClickForcedCheck: () => void;
  onClose: () => void;
  open: boolean;
  transformOrigin?: PositionOrigin;
```

Reply:
for   horizontal: 'left it s a string not enum , so ts is complaining

Reply:
ts is complaining because you created enums. But it's ok. Keep the enums. 
Just create a common interface to hold your `PositionOrigin`

Reply:
no , i did the same way that you mentioned , ok for PositionOrigin